sound/soc/sof/ipc4-topology.c

```diff
@@ -943,6 +949,9 @@ static int sof_ipc4_widget_setup_comp_process(struct snd_sof_widget *swidget)
 
 	sof_ipc4_widget_update_kcontrol_module_id(swidget);
 
+	/* set pipeline core mask to keep track of the core the module is scheduled to run on */
+	spipe->core_mask |= BIT(swidget->core);
```

Comment:
question: do we check if the cores used by topology are compatible with the hardware definitions?

Just being paranoid, e.g. if we use a topology defined for 4 cores on a SKU that only has 3.

Reply:
not during topology parsing but we do check before trying to do a core_get op